Rules/UseDeclaredVarsMoreThanAssignments.cs

```diff
@@ -208,34 +208,78 @@ private IEnumerable<DiagnosticRecord> AnalyzeScriptBlockAst(ScriptBlockAst scrip
+            ScriptBlockAst scriptBlockAst,
+            Dictionary<string, AssignmentStatementAst> assignmentsDictionary_OrdinalIgnoreCase)
+        {
             var getVariableCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("Get-Variable");
             IEnumerable<Ast> getVariableCommandAsts = scriptBlockAst.FindAll(testAst => testAst is CommandAst commandAst &&
                 getVariableCmdletNamesAndAliases.Contains(commandAst.GetCommandName(), StringComparer.OrdinalIgnoreCase), true);
+
             foreach (CommandAst getVariableCommandAst in getVariableCommandAsts)
             {
                 var commandElements = getVariableCommandAst.CommandElements.ToList();
-                // The following extracts the variable name only in the simplest possibe case 'Get-Variable variableName'
-                if (commandElements.Count == 2 && commandElements[1] is StringConstantExpressionAst constantExpressionAst)
+                // The following extracts the variable name(s) only in the simplest possible usage of Get-Variable.
+                // Usage of a named parameter and an array of variables is accounted for though.
+                if (commandElements.Count < 2 || commandElements.Count > 3) { continue; }
+
+                var commandElementAstOfVariableName = commandElements[commandElements.Count - 1];
+                if (commandElements.Count == 3)
+                {
+                    var commandParameterAst = commandElements[1] as CommandParameterAst;
+                    if (commandParameterAst == null) { continue; }
+                    // Check if the named parameter -Name is used (PowerShell does not need the full
+                    // parameter name and there is no other parameter of Get-Variable starting with n).
+                    if (!commandParameterAst.ParameterName.StartsWith("n", StringComparison.OrdinalIgnoreCase))
+                    {
+                        continue;
+                    }
+                }
+
+                if (commandElementAstOfVariableName is StringConstantExpressionAst constantExpressionAst)
+                {
+                    MarkVariableAsAssigned(constantExpressionAst.Value);
+                    continue;
+                }
+
+                var arrayLiteralAst = commandElementAstOfVariableName as ArrayLiteralAst;
+                if (arrayLiteralAst == null) { continue; }
+                foreach (var expressionAst in arrayLiteralAst.Elements)
+                {
+                    if (expressionAst is StringConstantExpressionAst constantExpressionAstOfArray)
+                    {
+                        MarkVariableAsAssigned(constantExpressionAstOfArray.Value);
+                    }
+                }
+
+                void MarkVariableAsAssigned(string variableName)
                 {
-                    var variableName = constantExpressionAst.Value;
                     if (assignmentsDictionary_OrdinalIgnoreCase.ContainsKey(variableName))

```

Comment:
According to [the documentation for `Remove`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.remove?view=netstandard-2.0), the first check is unnecessary

Reply:
In which case the local method is not needed -- we can directly remove the key from the dictionary in one line